data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/core/validation/PluginError.java

```diff
@@ -1,4 +1,4 @@
-package org.opensearch.dataprepper.validation;
+package org.opensearch.dataprepper.core.validation;
```

Comment:
These model classes should probably remain in `org.opensearch.dataprepper.validation`.

This includes: `PluginError`, `PluginErrorCollector`, and `PluginErrorsHandler`.

I think keeping `LoggingPluginErrorsHandler` in the new `.core.validation` package makes sense as this is a concrete implementation.

Reply:
I am unable to run my test pipeline if I put them back in their original package. I think, because these are tagged with `@Named` annotation and if they are not part of the scanning package, their injection is failing 

`jar:file:/Users/gandheaz/opensource-projects/san81/data-prepper/data-prepper-plugin-framework/build/libs/data-prepper-plugin-framework-2.10.0-SNAPSHOT.jar!/org/opensearch/dataprepper/plugin/ExtensionLoader.class]: Unsatisfied dependency expressed through constructor parameter 3; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.opensearch.dataprepper.validation.PluginErrorCollector' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}`